HPX fails to link if the MPI parcelport is enabled

## Expected Behavior

HPX should link successfully.

## Actual Behavior

Linking HPX fails with the following linker errors:

```
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Op::Init(void (*)(void const*, void*, int, MPI::Datatype const&), bool)':
version.cpp:(.text._ZN3MPI2Op4InitEPFvPKvPviRKNS_8DatatypeEEb[_ZN3MPI2Op4InitEPFvPKvPviRKNS_8DatatypeEEb]+0x19): undefined reference to `ompi_mpi_cxx_op_intercept'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Intracomm::Clone() const':
version.cpp:(.text._ZNK3MPI9Intracomm5CloneEv[_ZNK3MPI9Intracomm5CloneEv]+0x3f): undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Graphcomm::Clone() const':
version.cpp:(.text._ZNK3MPI9Graphcomm5CloneEv[_ZNK3MPI9Graphcomm5CloneEv]+0x38): undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Cartcomm::Sub(bool const*) const':
version.cpp:(.text._ZNK3MPI8Cartcomm3SubEPKb[_ZNK3MPI8Cartcomm3SubEPKb]+0xa4): undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Intracomm::Create_graph(int, int const*, int const*, bool) const':
version.cpp:(.text._ZNK3MPI9Intracomm12Create_graphEiPKiS2_b[_ZNK3MPI9Intracomm12Create_graphEiPKiS2_b]+0x3b): undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o: in function `MPI::Cartcomm::Clone() const':
version.cpp:(.text._ZNK3MPI8Cartcomm5CloneEv[_ZNK3MPI8Cartcomm5CloneEv]+0x38): undefined reference to `MPI::Comm::Comm()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:version.cpp:(.text._ZNK3MPI9Intracomm11Create_cartEiPKiPKbb[_ZNK3MPI9Intracomm11Create_cartEiPKiPKbb]+0xae): more undefined references to `MPI::Comm::Comm()' follow
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:(.data.rel.ro._ZTVN3MPI8DatatypeE[_ZTVN3MPI8DatatypeE]+0x78): undefined reference to `MPI::Datatype::Free()'
/usr/bin/ld: CMakeFiles/hpx.dir/version.cpp.o:(.data.rel.ro._ZTVN3MPI3WinE[_ZTVN3MPI3WinE]+0x48): undefined reference to `MPI::Win::Free()'

```

## Steps to Reproduce the Problem

Compile HPX with CMake >= 3.10 and OpenMPI with HPX_WITH_PARCELPORT_MPI enabled.

## Specifications

  - HPX Version: 1.2.0 (commit: e7d1ae4c5ba91b841fd2fa7fed99b712d0270cfe)
  - Platform (compiler, OS): Arch Linux (GCC 8.2.1, OpenMPI 3.1.2, CMake 3.12.2)

I have a, rather ugly, workaround for the linker errors. If the issue should be fixed immediately I could submit a PR containing the workaround. Alternatively, I am currently working on a proper fix. The root cause seems to be a problem with the scope of several CMake variables, exacerbated by the use of macros. It should be possible to get rid of both the macros and most of variables by using modern CMake targets and user-defined target properties. I have made good experiences using this approach in my current project.

--- a/plugins/parcelport/mpi/CMakeLists.txt
+++ b/plugins/parcelport/mpi/CMakeLists.txt
@@ -10,7 +10,9 @@ include(HPX_AddLibrary)
 # Decide whether to use the MPI based parcelport
 ################################################################################
 if(HPX_WITH_PARCELPORT_MPI)
+  set(MPI_CXX_SKIP_MPICXX TRUE)
   find_package(MPI)
+
   if(NOT MPI_CXX_FOUND)
     hpx_error("MPI could not be found and HPX_WITH_PARCELPORT_MPI=On, please specify MPI_CXX_COMPILER to point to a working MPI C++ compiler for your platform")
   endif()

--- a/tests/headers/CMakeLists.txt
+++ b/tests/headers/CMakeLists.txt
@@ -35,7 +35,9 @@ endif()
 # add the MPI include path here, because for the main library, it's only
 # added for the plugin.
 if(HPX_WITH_PARCELPORT_MPI)
+  set(MPI_CXX_SKIP_MPICXX TRUE)
   find_package(MPI)
+
   if(MPI_CXX_INCLUDE_PATH)
     include_directories(${MPI_CXX_INCLUDE_PATH})
   endif()

